### PR TITLE
feat(auth): get plan eligibility

### PR DESCRIPTION
**Pull request by julianpoy**
## Because

There isn't a great way to currently get your eligibility for a given plan outside of a 1-to-1 upgrade from plan A to B.

## This pull request

This endpoint provides a way to check your eligibility path for a given plan, and will return an eligibility status of:
- create (user can create a brand new subscription to this plan)
- upgrade (user is subscribed to a plan that can be upgraded to this plan)
- downgrade (user is subscribed to a plan that can be downgraded to this plan
- blocked_iap (user is subscribed to a plan via IAP that includes a productSet that conflicts with this plan and therefore cannot subscribe to this plan)
- invalid (configuration issue)

## Issue that this pull request solves

Closes: FXA-5479

## Checklist

_Put an `x` in the boxes that apply_

- [x] My commit is GPG signed.
- [x] If applicable, I have modified or added tests which pass locally.
- [x] I have added necessary documentation (if appropriate).
- [ ] I have verified that my changes render correctly in RTL (if appropriate).

### CODE REVIEW

File: packages/fxa-auth-server/lib/payments/capability.ts
Comment (StaberindeZA) [inline packages/fxa-auth-server/lib/payments/capability.ts:317]: How easy would it be to split the iapRoadblock logic into it's own function?

Reason I'm asking is we'll probably have to update the logic where we currently use`getSubscriptionUpdateEligibility` so that it also does the IAP checks.

I don't think we need to do it as part of this ticket. Do you know if there is a follow up ticket for that? Might need to create one otherwise if necessary.

**Review by StaberindeZA — COMMENTED** [review]
Just some minor initial review comments. I'd still like to test it locally once I have everything setup.

File: packages/fxa-auth-server/lib/payments/capability.ts
Comment (StaberindeZA) [inline packages/fxa-auth-server/lib/payments/capability.ts:285]: If it's a quick change, could we put all 3 `await` calls into a `Promise.all`? This shouldn't hold up merging the code though.

**Review by StaberindeZA — APPROVED** [review]
r+ wc. 

Before merging, please update the docs for the new route. Otherwise, I had 1 more comment for an optional change.

File: packages/fxa-auth-server/lib/payments/capability.ts
Comment (julianpoy) [reply packages/fxa-auth-server/lib/payments/capability.ts]: Added, as well as cleaned up the mapping a little bit. Not too big of a change but still meaningful. Thanks!

File: packages/fxa-auth-server/lib/payments/capability.ts
Comment (julianpoy) [reply packages/fxa-auth-server/lib/payments/capability.ts:319]: Spoke about this in person with @StaberindeZA but documenting here for historical reasons.

`getSubscriptionUpdateEligibility` currently doesn't have the information needed to make a decision WRT IAP subs, since IAP roadblock needs to consider all subscriptions under a given user rather than two individual plans. Adding IAP roadblock to the `getSubscriptionUpdateEligibility` util would fairly significantly increase it's size and complexity.

I believe the path forward here is to use `getPlanEligibility` in all locations that currently use `getSubscriptionUpdateEligibility`, and move `getSubscriptionUpdateEligibility` out of fxa-shared since it's currently used by the frontend. Meghan is already working on the latter part of that, and I'll create a ticket for the former.

File: packages/fxa-auth-server/lib/routes/subscriptions/mozilla.ts
Comment (julianpoy) [reply packages/fxa-auth-server/lib/routes/subscriptions/mozilla.ts]: Whoops, yep. Thanks - didn't see that was a thing.

**Review by bbangert — COMMENTED** [review]
There seems to be a substantial amount of conditions in the function that are not checked. It would probably be good to refactor the function down a bit. We should probably return a better error message if the getPlanElegibility function errors when trying to run the 3 promises.

File: packages/fxa-auth-server/lib/payments/capability.ts
Comment (julianpoy) [reply packages/fxa-auth-server/lib/payments/capability.ts:261]: Got it, I know reduce can be a divisive thing. I can remove here, though would love to see a linting rule to enforce this stylistic rule.

File: packages/fxa-auth-server/lib/payments/capability.ts
Comment (julianpoy) [reply packages/fxa-auth-server/lib/payments/capability.ts:273]: Gotcha - I believe there should be a linter rule we can enable to enforce this as well.

File: packages/fxa-auth-server/lib/payments/capability.ts
Comment (julianpoy) [reply packages/fxa-auth-server/lib/payments/capability.ts:356]: Happy to break this down farther. I would prefer to see this exist in it's own file but I was told `capability.ts` was the place this should go. I don't know that any individual parts of this would benefit from existing on their own, but breaking it up would still improve readability.

What are your thoughts WRT where this should live? Is my desire to move it out of this file completely reasonable?

**Pull request merged**
